splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsOperation.java

```diff
@@ -0,0 +1,155 @@
+import com.splicemachine.pipeline.Exceptions;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by jleach on 2/27/17.
+ */
+public class StatisticsOperation extends SpliceBaseOperation {
+    private static Logger LOG=Logger.getLogger(StatisticsOperation.class);
+
+    protected ScanSetBuilder scanSetBuilder;
+    protected String scope;
+
+    public StatisticsOperation(ScanSetBuilder scanSetBuilder, String scope, Activation activation) {
+        this.scanSetBuilder = scanSetBuilder;
+        this.scope = scope;
+        this.activation = activation;
+    }
+
+    @Override
+    public DataSet<LocatedRow> getDataSet(DataSetProcessor dsp) throws StandardException {
+        dsp.setSchedulerPool("admin");
+        try {
+            DataSet statsDataSet;
+            if (scanSetBuilder.getStoredAs() != null) {
+                ScanSetBuilder builder = scanSetBuilder;
+                String storedAs = scanSetBuilder.getStoredAs();
+                if (storedAs.equals("T"))
+                    statsDataSet = dsp.readTextFile(null, builder.getLocation(), builder.getDelimited(), null, builder.getColumnPositionMap(), null, null, null, builder.getTemplate());
+                else if (storedAs.equals("P"))
+                    statsDataSet = dsp.readParquetFile(builder.getColumnPositionMap(), builder.getPartitionByColumnMap() , builder.getLocation(), null, null, null, builder.getTemplate());
+                else if (storedAs.equals("O"))
+                    statsDataSet = dsp.readORCFile(builder.getColumnPositionMap(), builder.getPartitionByColumnMap(), builder.getLocation(), null, null, null, builder.getTemplate());
+                else {
+                    throw new UnsupportedOperationException("storedAs Type not supported -> " + storedAs);
+                }
+            } else {
+                statsDataSet = scanSetBuilder.buildDataSet(scope);
+            }
+            return statsDataSet
+                    .mapPartitions(
+                            new StatisticsFlatMapFunction(scanSetBuilder.getBaseTableConglomId(), scanSetBuilder.getColumnPositionMap(), scanSetBuilder.getTemplate()));
+        } catch (StandardException se) {
+            throw se;
+        }
+
+    }
+
+    @Override
+    public String getName() {
+        return "StatisticsCollection";
+    }
+
+    @Override
+    public int[] getRootAccessedCols(long tableNumber) throws StandardException {
+        return new int[0];
+    }
+
+    @Override
+    public boolean isReferencingTable(long tableNumber) {
+        return false;
+    }
+
+    @Override
+    public String prettyPrint(int indentLevel) {
+        return "statistics " + scope;
+    }
+
+    @Override
+    public List<SpliceOperation> getSubOperations() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void openCore() throws StandardException{
```

Comment:
@snowspeeder This should submit the StatisticsOperation to the master, is that not happening?